### PR TITLE
refactor: removing redundancy by creating new macro for asserting (#425)

### PR DESCRIPTION
**Pull request by subhakundu**
In this PR, I have added new rules in assert_lint_err macro, to support common messages and hints.

Test done:
1. Pulled latest changes
2. cargo build --all-targets
3. cargo test

### CODE REVIEW

File: src/test_util.rs
Comment (magurotuna) [inline src/test_util.rs:81]:
```suggestion
    $(
      let errors = parse_err_test!($message, $hint, $test);
      let tester = $crate::test_util::LintErrTester::<$rule>::new(
        $src,
        errors,
        $filename,
      );
      tester.run();
    )*
```

This should look more beautiful 👍 (it is bad that rustfmt cannot do its job on macros)

File: src/test_util.rs
Comment (magurotuna) [inline src/test_util.rs:97]:
```suggestion
  ) => {
    assert_lint_err! {
      $rule,
      $message,
      $hint,
      filename: "deno_lint_err_test.ts",
      $($src: $test,)*
    }
  };
```
This is also about formatting. Since Rust codes in Deno adopt 2-space indent, it'd be better to follow it in a macro.

**Review by magurotuna — CHANGES_REQUESTED** [review]
Very nice 👍  a little nitpicking though.

File: src/test_util.rs
Comment (subhakundu) [reply src/test_util.rs:97]: Sure, it's my bad that I did not follow the convention. I will modify and update the PR. Thank you for suggesting the changes.

**Comment by subhakundu** [comment]
@magurotuna I have incorporated the changes and created two commits. Please let me know if I need to modify anything. Please let me know if I have made any mistake during updating PR.

File: src/rules/ban_ts_comment.rs
Comment (magurotuna) [inline src/rules/ban_ts_comment.rs:198]:
```suggestion
      ],
```

File: src/rules/ban_ts_comment.rs
Comment (magurotuna) [inline src/rules/ban_ts_comment.rs:225]:
```suggestion
      ],
```

File: src/rules/ban_ts_comment.rs
Comment (magurotuna) [inline src/rules/ban_ts_comment.rs:252]:
```suggestion
      ],
```

**Review by magurotuna — CHANGES_REQUESTED** [review]
I found other very minor problems. Otherwise looks good :)

**Comment by subhakundu** [comment]
Thank you @magurotuna for catching these. From next time onwards, I will be careful not to miss out these errors.

**Review by magurotuna — APPROVED** [review]
Perfect, LGTM 💯 
Before landing I have to leave a final review to someone in the core team. 
@bartlomieju @dsherret could you take a look?

**Review by bartlomieju — APPROVED** [review]
LGTM, thanks @subhakundu!

**Pull request merged**
